http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java

```diff
@@ -0,0 +1,119 @@
+package io.quarkus.ts.http.advanced.reactive;
+
+import static io.quarkus.ts.http.advanced.reactive.MediaTypeResource.APPLICATION_YAML;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+
+@Path("/city")
+public class CityResource {
+
+    private static final String YAML_FILE_PATH = "payload.yaml";
+
+    private final Logger logger = LoggerFactory.getLogger(CityResource.class);
+
+    private final ObjectMapper objectMapper = new YAMLMapper();
+
+    @Inject
+    private CityListWrapperSerializer serializer;
+
+    private List<City> cityList = new ArrayList<>();
+
+    @GET
+    @Produces(value = MediaType.APPLICATION_XML)
+    public CityListWrapper getCities() {
+        logger.info("Received request to getCities");
+
+        if (cityList.isEmpty()) {
+            cityList.add(new City("San Bernardino", "EEUU"));
+            cityList.add(new City("Brno", "Czech Republic"));
+            cityList.add(new City("Zaragoza", "Spain"));
+        }
+        return new CityListWrapper(cityList);
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_XML)
+    @Produces(value = MediaType.APPLICATION_XML)
+    public String createCity(String xml) {
+
+        logger.info("Received XML payload: {}", xml);
+
+        CityListDTO cityListDTO = serializer.fromXML(xml);
+
+        if (cityListDTO == null) {
+            logger.error("Error deserializing XML");
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("<error>Invalid XML payload</error>")
+                    .build().toString();
+        }
+
+        List<City> cities = cityListDTO.getCityList();
+
+        for (City city : cities) {
+            logger.info("Saving city {}", city);
+        }
+
+        String responseXML = serializer.toXML(cityListDTO);
+
+        return responseXML;
+    }
+
+    @POST
+    @Path("/cityYaml")
+    @Produces(MediaType.APPLICATION_JSON)
```

Comment:
Why is `handleYamlPostRequest` producing application/json?

Reply:
Why not? I mean the request already uses a YAML payload, and also in the @GET we already checked that it could be @Produce a Yaml here:    [ @Produces(APPLICATION_YAML)](https://github.com/quarkus-qe/quarkus-test-suite/blob/7db3158f3b330b551335d884f02c8bb0c6cc7f34/http/http-advanced-reactive/src/main/java/io/quarkus/ts/http/advanced/reactive/CityResource.java#L100)

Reply:
ok

Reply:
Tbh it's kinda confusing at first gaze - why not just remove `Yaml` from the path? If people come back to this test in the future, they'll be scratching their head too :]

Reply:
truth be told, I didn't want to argue, but I think name should reflect that yaml is only consumed, not produced. like `handleConsumedYamlPostRequest` (just better :-)).

I also think it is rather rare to consume YAML and produce JSON and there is no scenario AFAICT that tests both consume/produce YAML, but I'll respect @jcarranzan preference.

Reply:
Yes, looking at it again, you're right I will tweak it to change the path and the name to `handleConsumedYamlPostRequest`